### PR TITLE
[documentation] explain how to implement flags

### PR DESCRIPTION
**Pull request by trombik**

### CODE REVIEW

File: docs/How_Do_I/README.md [hunk 1]
```diff
@@ -773,3 +776,94 @@ only the number of changes that the role has made to the system.
...
+### Problem
+
+Often, a service provides a way to specify additional command line flags for
+the service command. Different platform has a different mechanism to manage the
+flags, some even provides extra variables under `/etc/sysconfig` for
+RedHat-variants, `/etc/default` for Debian-variants, and `/etc/rc.conf.d` for
+FreeBSD. OpenBSD has a similar mechanism, but uses a single file,
+`/etc/rc.conf.local`.
+
+### Solution
```
Comment (mitsururike) [inline docs/How_Do_I/README.md]: the codes explain everything but some explanation should be written before the codes,
such as
* *_flags_default keeps defaults provided by the upstream
* *_flags is for users
* merge *_flags_default and *_flags

Comment (trombik) [inline docs/How_Do_I/README.md]: fixed in 15d992a

File: docs/How_Do_I/README.md [hunk 2]
```diff
@@ -773,3 +776,118 @@ only the number of changes that the role has made to the system.
...
+
+Define `foo_flags` and `foo_flags_default` in `defalts/main.yml`. `foo_flags`
+is a dict for users to change the default, and `foo_flags_default` is a dict
+that contains platform's default variables and values. Merge them into
+`foo_flags_merged` so that `foo_flags` overrides `foo_flags_default` as early
```
Comment (mitsururike) [inline docs/How_Do_I/README.md]: `foo_flags` does NOT override `foo_flags_default`.

Comment (trombik) [inline docs/How_Do_I/README.md]: what do you mean? if it does not, why is it necessary?

Comment (mitsururike) [inline docs/How_Do_I/README.md]: `foo_flags overrides` means `foo_flags ignores foo_flags_default`.
they are merged and `overrides` is not an appropriate word.

Comment (trombik) [inline docs/How_Do_I/README.md]: fixed in c25c647

**Pull request merged**